\hypertarget{exception_8h}{}\doxysection{librapid/include/librapid/cuda/exception.h File Reference}
\label{exception_8h}\index{librapid/include/librapid/cuda/exception.h@{librapid/include/librapid/cuda/exception.h}}
{\ttfamily \#include $<$exception$>$}\newline
{\ttfamily \#include $<$iostream$>$}\newline
{\ttfamily \#include $<$stdexcept$>$}\newline
{\ttfamily \#include $<$stdlib.\+h$>$}\newline
{\ttfamily \#include $<$string$>$}\newline
{\ttfamily \#include $<$sstream$>$}\newline
\doxysubsection*{Classes}
\begin{DoxyCompactItemize}
\item 
class \mbox{\hyperlink{class_exception}{Exception$<$ Std\+\_\+\+Exception $>$}}
\end{DoxyCompactItemize}
\doxysubsection*{Macros}
\begin{DoxyCompactItemize}
\item 
\#define \mbox{\hyperlink{exception_8h_a14d84c56a2433cf0b894bf62cf8a5c31}{RUNTIME\+\_\+\+EXCEPTION}}(msg)~\mbox{\hyperlink{class_exception}{Exception}}$<$std\+::runtime\+\_\+error$>$\+::throw\+\_\+it(\+\_\+\+\_\+\+FILE\+\_\+\+\_\+, \+\_\+\+\_\+\+LINE\+\_\+\+\_\+, msg)
\begin{DoxyCompactList}\small\item\em Convenience macros. \end{DoxyCompactList}\item 
\#define \mbox{\hyperlink{exception_8h_a994cf587529580a7865cdf9a7016514a}{LOGIC\+\_\+\+EXCEPTION}}(msg)~\mbox{\hyperlink{class_exception}{Exception}}$<$std\+::logic\+\_\+error$>$\+::throw\+\_\+it(\+\_\+\+\_\+\+FILE\+\_\+\+\_\+, \+\_\+\+\_\+\+LINE\+\_\+\+\_\+, msg)
\begin{DoxyCompactList}\small\item\em Logic exception in program, e.\+g. an assert failed. \end{DoxyCompactList}\item 
\#define \mbox{\hyperlink{exception_8h_ad7375e2978467b3ca52550fc2d2356d6}{RANGE\+\_\+\+EXCEPTION}}(msg)~\mbox{\hyperlink{class_exception}{Exception}}$<$std\+::range\+\_\+error$>$\+::throw\+\_\+it(\+\_\+\+\_\+\+FILE\+\_\+\+\_\+, \+\_\+\+\_\+\+LINE\+\_\+\+\_\+, msg)
\begin{DoxyCompactList}\small\item\em Out of range exception. \end{DoxyCompactList}\end{DoxyCompactItemize}
\doxysubsection*{Functions}
\begin{DoxyCompactItemize}
\item 
{\footnotesize template$<$class Exception\+\_\+\+Typ $>$ }\\void \mbox{\hyperlink{exception_8h_a35bcf04d6841d8287e250ce19096e15d}{handle\+Exception}} (const Exception\+\_\+\+Typ \&ex)
\end{DoxyCompactItemize}


\doxysubsection{Macro Definition Documentation}
\mbox{\Hypertarget{exception_8h_a994cf587529580a7865cdf9a7016514a}\label{exception_8h_a994cf587529580a7865cdf9a7016514a}} 
\index{exception.h@{exception.h}!LOGIC\_EXCEPTION@{LOGIC\_EXCEPTION}}
\index{LOGIC\_EXCEPTION@{LOGIC\_EXCEPTION}!exception.h@{exception.h}}
\doxysubsubsection{\texorpdfstring{LOGIC\_EXCEPTION}{LOGIC\_EXCEPTION}}
{\footnotesize\ttfamily \#define LOGIC\+\_\+\+EXCEPTION(\begin{DoxyParamCaption}\item[{}]{msg }\end{DoxyParamCaption})~\mbox{\hyperlink{class_exception}{Exception}}$<$std\+::logic\+\_\+error$>$\+::throw\+\_\+it(\+\_\+\+\_\+\+FILE\+\_\+\+\_\+, \+\_\+\+\_\+\+LINE\+\_\+\+\_\+, msg)}



Logic exception in program, e.\+g. an assert failed. 



Definition at line \mbox{\hyperlink{exception_8h_source_l00087}{87}} of file \mbox{\hyperlink{exception_8h_source}{exception.\+h}}.

\mbox{\Hypertarget{exception_8h_ad7375e2978467b3ca52550fc2d2356d6}\label{exception_8h_ad7375e2978467b3ca52550fc2d2356d6}} 
\index{exception.h@{exception.h}!RANGE\_EXCEPTION@{RANGE\_EXCEPTION}}
\index{RANGE\_EXCEPTION@{RANGE\_EXCEPTION}!exception.h@{exception.h}}
\doxysubsubsection{\texorpdfstring{RANGE\_EXCEPTION}{RANGE\_EXCEPTION}}
{\footnotesize\ttfamily \#define RANGE\+\_\+\+EXCEPTION(\begin{DoxyParamCaption}\item[{}]{msg }\end{DoxyParamCaption})~\mbox{\hyperlink{class_exception}{Exception}}$<$std\+::range\+\_\+error$>$\+::throw\+\_\+it(\+\_\+\+\_\+\+FILE\+\_\+\+\_\+, \+\_\+\+\_\+\+LINE\+\_\+\+\_\+, msg)}



Out of range exception. 



Definition at line \mbox{\hyperlink{exception_8h_source_l00090}{90}} of file \mbox{\hyperlink{exception_8h_source}{exception.\+h}}.

\mbox{\Hypertarget{exception_8h_a14d84c56a2433cf0b894bf62cf8a5c31}\label{exception_8h_a14d84c56a2433cf0b894bf62cf8a5c31}} 
\index{exception.h@{exception.h}!RUNTIME\_EXCEPTION@{RUNTIME\_EXCEPTION}}
\index{RUNTIME\_EXCEPTION@{RUNTIME\_EXCEPTION}!exception.h@{exception.h}}
\doxysubsubsection{\texorpdfstring{RUNTIME\_EXCEPTION}{RUNTIME\_EXCEPTION}}
{\footnotesize\ttfamily \#define RUNTIME\+\_\+\+EXCEPTION(\begin{DoxyParamCaption}\item[{}]{msg }\end{DoxyParamCaption})~\mbox{\hyperlink{class_exception}{Exception}}$<$std\+::runtime\+\_\+error$>$\+::throw\+\_\+it(\+\_\+\+\_\+\+FILE\+\_\+\+\_\+, \+\_\+\+\_\+\+LINE\+\_\+\+\_\+, msg)}



Convenience macros. 

\mbox{\hyperlink{class_exception}{Exception}} caused by dynamic program behavior, e.\+g. file does not exist 

Definition at line \mbox{\hyperlink{exception_8h_source_l00084}{84}} of file \mbox{\hyperlink{exception_8h_source}{exception.\+h}}.



\doxysubsection{Function Documentation}
\mbox{\Hypertarget{exception_8h_a35bcf04d6841d8287e250ce19096e15d}\label{exception_8h_a35bcf04d6841d8287e250ce19096e15d}} 
\index{exception.h@{exception.h}!handleException@{handleException}}
\index{handleException@{handleException}!exception.h@{exception.h}}
\doxysubsubsection{\texorpdfstring{handleException()}{handleException()}}
{\footnotesize\ttfamily template$<$class Exception\+\_\+\+Typ $>$ \\
void handle\+Exception (\begin{DoxyParamCaption}\item[{const Exception\+\_\+\+Typ \&}]{ex }\end{DoxyParamCaption})\hspace{0.3cm}{\ttfamily [inline]}}

\mbox{\hyperlink{class_exception}{Exception}} handler function for arbitrary exceptions 
\begin{DoxyParams}{Parameters}
{\em ex} & exception to handle \\
\hline
\end{DoxyParams}


Definition at line \mbox{\hyperlink{exception_8h_source_l00075}{75}} of file \mbox{\hyperlink{exception_8h_source}{exception.\+h}}.

